<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:include filename="inertia.xacro"/>

    <link name="base_footprint"/>

    <link name="base_link">
        <xacro:inertial_cylinder mass="0.1" radius="0.15" length="0.05">
            <origin xyz= "0 0 0.05" />
        </xacro:inertial_cylinder>
        <visual>
            <origin xyz="0.0 0.0 0.025"/>
            <geometry>
                <cylinder radius="0.15" length="0.05"/>/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.0 0.0 0.025"/>
            <geometry>
                <cylinder radius="0.15" length="0.05"/>/>
            </geometry>
        </collision>
    </link>

    <joint name= "base_joint" type= "fixed">
        <parent link="base_footprint"/>
        <child link="base_link"/>
        <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
        <axis xyz="0 0 0"/>
    </joint>

    <link name="link1">
        <xacro:inertial_cylinder mass="0.05" radius="0.05" length="0.475">
            <origin xyz= "0 0 0" rpy="0 0 0" />
        </xacro:inertial_cylinder>
        <visual>
            <origin xyz="0 0 0.0"/>
            <geometry>
                <cylinder radius="0.05" length="0.475"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0 0 0.0"/>
            <geometry>
                <cylinder radius="0.05" length="0.475"/>
            </geometry>
        </collision>
    </link>

    <joint name= "link1_joint" type= "fixed">
        <parent link="base_link"/>
        <child link="link1"/>
        <origin xyz="0.0 0.0 0.2875"/>
        <axis xyz="0 0 0"/>
    </joint>

    <link name="link2">
        <xacro:inertial_box mass="0.5" x="0.5" y="0.1" z="0.1">
            <origin xyz= "0.2 0 0.0" />
        </xacro:inertial_box>
        <visual name="">
            <origin xyz="0.2 0.0 0.0"/>
            <geometry>
                <box size="0.5 0.1 0.05"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.2 0.0 0.0"/>
            <geometry>
                <box size="0.5 0.1 0.05"/>
            </geometry>
        </collision>
    </link>

    <joint name="link2_joint" type="revolute">
        <parent link="base_link"/>
        <child link="link2"/>
        <origin xyz="0.0 0.0 0.55" rpy="0.0 0.0 0.0"/>
        <axis xyz="0 0 1"/>
        <limit lower="-1.5708" upper="1.5708" effort="0" velocity="0"/>
    </joint>

    <link name="link3">
        <xacro:inertial_box mass="0.5" x="0.5" y="0.1" z="0.1">
            <origin xyz= "0.15 0 0.075" rpy="0 0 0" />
        </xacro:inertial_box>
        <visual name="">
            <origin xyz="0.2 0.0 0.0"/>
            <geometry>
                <box size="0.5 0.1 0.05"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.2 0.0 0.0"/>
            <geometry>
                <box size="0.5 0.1 0.05"/>
            </geometry>
        </collision>
    </link>

    <joint name="link3_joint" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin xyz="0.425 0.0 0.05" />
        <axis xyz="0 0 1"/>
        <limit lower="-1.5708" upper="1.5708" effort="0" velocity="0"/>
    </joint>

    <link name="effector">
        <xacro:inertial_cylinder mass="0.1" radius="0.05" length="0.22">
            <origin xyz= "0 0 0" rpy="0 0 0" />
        </xacro:inertial_cylinder>
        <visual>
            <origin xyz="0.0 0.0 0.11"/>
            <geometry>
                <cylinder radius="0.05" length="0.22"/>
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.0 0.0 0.11"/>
            <geometry>
                <cylinder radius="0.05" length="0.22"/>
            </geometry>
        </collision>
    </link>

    <joint name="effector_prismatic_joint" type="prismatic">
        <parent link="link3"/>
        <child link="effector"/>
        <origin xyz="0.345 0.0 -0.110" />
        <axis xyz="0 0 -1"/>
        <limit lower="0.0" upper="0.1" effort="0" velocity="0"/>
    </joint>
</robot>